File: exercise1/src/main/java/exercise1/data/CustomerDAO.java
package exercise1.data;

import java.util.Collection;
import java.util.List;

import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.data.domain.Pageable;

public interface CustomerDAO extends PagingAndSortingRepository<CustomerEntity, String> {

	public List<CustomerEntity> findAllByEmailDomain(@Param("emailDomain") String emailDomain, Pageable pageable);

	public List<CustomerEntity> findAllByYear(@Param("year") int year, Pageable pageable);
//	public List<CustomerEntity> findAllByBirthdateYear(@Param("birthdateYear") int birthdateYear, Pageable pageable);

	public List<CustomerEntity> findAllByEmailIn(@Param("email") Collection<String> emails, Pageable pageable);

}

File: New folder/Demo - Kotlin/src/main/kotlin/demo/CustomerDAO.kt
package demo

import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.data.repository.query.Param


interface CustomerDAO : PagingAndSortingRepository<CustomerEntity?, String?> {
    fun findAllByEmailDomain(@Param("emailDomain") emailDomain: String?, pageable: Pageable?): List<CustomerEntity?>?
    fun findAllByYear(@Param("year") year: Int, pageable: Pageable?): List<CustomerEntity?>?

    //	public List<CustomerEntity> findAllByBirthdateYear(@Param("birthdateYear") int birthdateYear, Pageable pageable);
    fun findAllByEmailIn(@Param("email") emails: Collection<String?>?, pageable: Pageable?): List<CustomerEntity?>?
}
